### PR TITLE
add post-http service

### PR DESCRIPTION
**Pull request by casterweb**

### CODE REVIEW

File: src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.html [hunk 1]
```diff
@@ -1,6 +1,18 @@
 <h1>Assign test setup</h1>
 <div mat-dialog-content>
-  <p><mat-checkbox formControlName="highpoint">High priority</mat-checkbox></p>
+  <p><mat-checkbox>High priority</mat-checkbox></p>
+  <h2>Select level</h2>
```
Comment (RomPavel) [inline src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.html:4]: Ask @ExxZzyy, we have common component for that

File: src/app/models/hr-profile/services/post-http.service.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { ApiAssignTest } from '../../../../interfaces/user.interfaces';
+import { PostAssignTest, RefreshUserDataUrl } from '../../../../constants/route-constant';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PostHttpService {
```
Comment (RomPavel) [inline src/app/models/hr-profile/services/post-http.service.ts:10]: Lets name it like AssignTestHttpService

File: src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.ts [hunk 3]
```diff
@@ -18,5 +19,19 @@ export class HrProfileDialogComponent {
     return day > this.currentDay.getDate();
   };
 
-  constructor(@Inject(MAT_DIALOG_DATA) public data: DialogData) {}
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    private readonly postHttpService: PostHttpService,
+  ) {}
+
+  assignTest() {
```
Comment (Inunor) [inline src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.ts:27]: Maybe you will have to store some posts data in the future, so don't forget to move this post http service call to posts store service and call posts store service instead

**Review by Inunor — APPROVED** [review]

File: src/constants/route-constant.ts [hunk 4]
```diff
@@ -19,6 +19,6 @@ export enum SidebarMenu {
   manage = 'Manage',
   statistics = 'Statistics',
 }
-
+export const PostAssignTest = 'http://elevel-001-site1.btempurl.com/api/Test/assign';
```
Comment (Inunor) [inline src/constants/route-constant.ts:22]: Better to move this const to your hr module

File: src/interfaces/user.interfaces.ts [hunk 5]
```diff
@@ -20,3 +20,10 @@ export interface UserToken {
   sub: string;
   uid: string;
 }
+
```
Comment (Inunor) [inline src/interfaces/user.interfaces.ts:23]: Same for this interface, I think you need it only inside your hr module

Comment (Inunor) [reply src/interfaces/user.interfaces.ts:23]: It isn't global I mean

**Pull request merged**